### DISCUSSION TITLE
"Undefined symbol" error on project using Package with TCA dependency

### DISCUSSION THREAD

**Discussion by WedgeSparda** [pointfreeco/swift-composable-architecture#2127] [Q&A]
Hi, I'm migrating our base code, formed by different local Frameworks into local Packages. Until now I was able to transform the bottom layer framework (with some Foundation extension and helpers, let's called `MyFoundation`) without problem. 

Now I'm working on the next layer called `MyCore`, a framework which has `MyFoundation` as a local dependency, and also some remote dependencies, like `ComposableArchitecture`.

I managed to create the `MyCore` package and it compiles without problem. This is the `Package.swift`, as you can see I'm using `prerelease/1.0`

```swift
import PackageDescription

let package = Package(
    name: "MyCore",
    defaultLocalization: "en",
    platforms: [
        .iOS(.v14),
        .watchOS(.v7)
    ],
    products: [
        .library(
            name: "MyCore",
            targets: [
                "MyCore",
                "MyCoreMocks"
            ]
        )
    ],
    dependencies: [
        .package(path: "../MyFoundation"),
        .package(url: "https://github.com/pointfreeco/swift-composable-architecture", branch: "prerelease/1.0"),
        // ... more external packages
    ],
    targets: [
        .target(
            name: "MyCore",
            dependencies: [
                .product(name: "MyFoundation", package: "MyFoundation"),
                .product(name: "ComposableArchitecture", package: "swift-composable-architecture"),
               // ... more dependencies from external packages
            ]
        ),
        .target(
            name: "MyCoreMocks",
            dependencies: ["MyCore"]
        )
    ]
)
```

As I said `MyCore` compiles without problem, but problem comes when I compile any of the frameworks using the `MyCore` package as dependency. I'm getting this errors:

```
Undefined symbols for architecture arm64:
  "_$s12Dependencies10DependencyV12wrappedValuexvg", referenced from:
      (...)
  "_$s12Dependencies10DependencyVMn", referenced from:
      (...)
  "_$s12Dependencies10DependencyV_4file0C2ID4lineACyxGs7KeyPathCyAA0B6ValuesVxG_s12StaticStringVANSutcfC", referenced from: 
      (...)
  "_$s12Dependencies13DependencyKeyMp", referenced from:
      (...)
  "_$s12Dependencies16DependencyValuesVMn", referenced from:
      _symbolic _____ 12Dependencies16DependencyValuesV in CabiCore.o
  "_$s12Dependencies16DependencyValuesVy5ValueQzxm_s12StaticStringVAGSutcAA04TestB3KeyRzluig", referenced from:
      (...)
  "_$s12Dependencies16DependencyValuesVy5ValueQzxm_s12StaticStringVAGSutcAA04TestB3KeyRzluis", referenced from:
      (...)
  "_$s12Dependencies17TestDependencyKeyMp", referenced from:
     (...)
ld: symbol(s) not found for architecture arm64
```
All the lines on the `referenced from` are from types using the `@Dependency` annotation, obviously.

I'm not sure why is this happening. I tried linking `ComposableArchitecture` to the frameworks using `MyCore` but didn't work.

What I'm missing?

Thanks for your time.

**Comment by mbrandonw** [comment]
Hi @WedgeSparda, I'm not sure how to diagnose this without a project that reproduces the problem. Can you try creating a project from scratch that mimics your project's structure, and see if it still happens?

**Comment by WedgeSparda** [comment]
Hi @mbrandonw I managed to create this demo project

https://github.com/WedgeSparda/TestSPMDependencyError

But for that project everything is working as expected. Maybe there is some build setting on my main project causing this issue but not sure which one yet. Anyway, in this demo project you can see basically whay I'm trying to do in my main project.

**Reply by mbrandonw** [reply]
Yeah hard to say. It does sound like something may be misconfigured in your other project.

**Comment by StefanCosminR** [comment]
This happened a few times for me as well, but it was just an Xcode / Swift bug in my case. Cleaning the build folder always fixes the problem

**Reply by WedgeSparda** [reply]
@StefanCosminR I tried that too, with no luck
